src/Map.js

```diff
@@ -31,14 +31,16 @@ class Map extends Component {
     if (this.hasPropChanged('bounds', prevProps)) {
       this.state.map.fitBounds(this.props.bounds);
     }
-
-    addListenersFromProps(this.state.map, this.props, {
-      filter: ({ prop }) => !prevProps[prop],
-    });
-
-    removeListenersFromProps(this.state.map, prevProps, {
-      filter: ({ prop }) => !this.props[prop],
-    });
+    
+    if( prevProps !== this.props) {
+      removeListenersFromProps(this.state.map, prevProps, {
+        filter: ({ prop }) => !this.props[prop],
+      });
+
+      addListenersFromProps(this.state.map, this.props, {
+        filter: ({ prop }) => !prevProps[prop],
+      });
+    }
```

Comment:
I can't see why this change will affect any behavior. Both `removeListenersFromProps` and `addListenersFromProps` are filtering out so they only receives props that has changed. So this added if statement is redundant.